ADBDEV-4726-68 Remove redundant null-check for planstate

dnskvlnk opened the pull request:
Remove redundant null-check for planstate

At this point planstate is always not NULL, so I removed the redundant planstate
null-check

RekGRpth commented on src/backend/executor/nodeSubplan.c:
```diff
@@ -965,21 +965,26 @@ ExecSetParamPlan(SubPlanState *node, ExprContext *econtext, QueryDesc *queryDesc
+	savepeakspace = MemoryContextGetPeakSpace(planstate->state->es_query_cxt);
+
 	if (Gp_role == GP_ROLE_DISPATCH &&
-		planstate != NULL &&
```
I think you should have just removed this condition.

dnskvlnk replied:
Done

RekGRpth replied:
Only remove condition without other changes.

dnskvlnk replied:
Why won't we check planstate at least with Assert?

RekGRpth replied:
Why, if it is definitely not equal to NULL?

dnskvlnk replied:
Can you guarantee that node can't have invalid planstate?

RekGRpth replied:
Yes, it is dereferenced at the very beginning.

dnskvlnk replied:
Yes, that's what I'm talking about. Can you guarantee that it can't be null here?
https://github.com/arenadata/gpdb/blob/c4f634eff2d5f2ee6483facb4b5e38e6c8c97013/src/backend/executor/nodeSubplan.c#L966

RekGRpth replied:
Well, it’s been dereferencing for 21 years already!

dnskvlnk replied:
Done

Commit:
Remove redundant null-check for planstate

At this point planstate is always not NULL, so I removed the redundant planstate
null-check
